src/applications/MachineShopSimulator.java

```diff
@@ -85,21 +87,10 @@ private static void setMachineChangeOverTimes(SimulationSpecification specificat
 
     private static void setUpJobs(SimulationSpecification specification) {
         // input the jobs
-        Job theJob;
-        for (int i = 1; i <= specification.getNumJobs(); i++) {
-            int tasks = specification.jobs[i].numTasks;
-            int firstMachine = 0; // machine for first task
-
-            // create the job
-            theJob = new Job(i);
-            for (int j = 1; j <= tasks; j++) {
-                int theMachine = specification.jobs[i].getSpecificationsForTasks()[2*(j-1)+1];
-                int theTaskTime = specification.jobs[i].getSpecificationsForTasks()[2*(j-1)+2];
-                if (j == 1)
-                    firstMachine = theMachine; // job's first machine
-                theJob.addTask(theMachine, theTaskTime); // add to
-            } // task queue
-            machine[firstMachine].getJobQ().put(theJob);
+        for (int i = 0; i < specification.getNumJobs(); i++) {
+            LinkedQueue jobTasks = specification.jobs[i].getTaskQ();
+            int firstMachine = ((Task)jobTasks.getFrontElement()).getMachine();
+            machine[firstMachine].getJobQ().put(specification.jobs[i]);

```

Comment:
Helps a lot. But, there seems to be a missed opportunity for some message chain reduction. I know it's not your primary smell though.

Reply:
I think you shorten a code by removing multi use int[] and removing deep copy functionality.  It becomes easy to read. However, there are few message chain in this code. I think it is equivalent exchange, so you can leave as is.